pcdsdevices/analog_signals.py

```diff
@@ -61,6 +61,33 @@ class Acromag(BaseInterface, Device):
     tab_component_names = True
 
 
+def acromag_factory_func(prefix, channel, *, signal_class=None, name=None,
```

Comment:
I think the code looks great. If Travis was ready I'd probably just hit merge.

I'm singling out this line to discuss opinions on our naming for factory functions. All of other other factory functions are pretending to be classes (e.g. `def Attenuator():`, `def Motor():`, this one is very honest about what it is doing. I like an honest function, I'm wondering if we want to take the lead from this kind of naming scheme and apply it elsewhere in the library in another (low priority) PR? Or do people like the magical classnamed factory functions? To be clear, I also think there is a lot to like about a factory function named, say, `AcromagChannel`

Reply:
Yeah great point, I went back and forth with the naming on this one, and I also took it out from the `device_types` as well - but maybe I should add it back in? (in happi qs_db for `device_class` for acromag entry creation, I just give directly the `pcdsdevices.analog_signal.acromag_factory_func` - is this ok?).
I can go back to using the `AcromagChannel` for this function too :) - might make it clearer?

Reply:
I personally like the "pretend to be class" format instead because, to the user, it essentially is a class. To be more honest, the docstring should be very clear about what it actually is like this factory function's is. I vote for `AcromagChannel`

Reply:
Ok should I switch it back to `AcromagChannel` - and should I also import it to the `device_types`?

Reply:
Since I expect it'll be used pretty commonly, yes on the `device_types`. Please also add a line in `docs/source/epics_types.rst` so that it gets included on that docs page.

As for the name change, please don't do it just because I said so. Let's get at least @klauer's opinion first, maybe @silkenelson's too

Reply:
I would expect an "Acromag" to be the whole system that can contain an ao card, an ai card, ...with the respective channels. If we are talking a single channel, I'd like to have "Channel" in the name (or "ch", "chn" ...)

Reply:
That's a good point. If we do go with the function format, this should be changed (although that is so long ☹️):
```suggestion
def acromag_channel_factory_func(prefix, channel, *, signal_class=None, name=None,
```
@silkenelson, what is your opinion on function versus class format for naming factory functions? `acromag_channel_factory_func` vs `AcromagChannel`

Reply:
The factory function name is obvious in its intent, a big point in its favor.
The class-like behavior of `AcromagChannel` is what a user might expect. This breaks down when you do `isinstance(obj, AcromagChannel)` (or expect any other class-like behavior).

For the above reasons, I'd tend toward the style-guide-approved, lowercase, underscored name. There are alternatives, as @ZLLentz did for `PytmcSignal` - but I don't believe adding that complexity is justified here.

Reply:
seems like we have some mixed opinions here...

Reply:
I agree that we seem to be in disagreement!
Let's double-dip for now using an `AcromagChannel = acromag_factory_function` alias maybe?

Reply:
I'm fine with going function format. I'm also happy to have the alias in there at least for now.